gh-actions/go/gotestfmt/gotestfmt

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# This script is a wrapper around gotestfmt. It expects the output of `go test -json`
+# on stdin. The output of failed tests is written to stdout, while the full output
+# (including successful tests) is written to a logfile.
+
+usage(){
+    echo "Usage: $0 [--logfile <logfile>] [-h|--help]"
+}
+
+# Parse options
+while [[ $# -gt 0 ]]; do
+    case "$1" in
+        --logfile)
+            LOGFILE="$2"
+            shift 2
+            ;;
+        -h|--help)
+            usage
+            exit 0
+            ;;
+        *)
+            echo "Unknown option: $1"
+            usage
+            exit 1
+            ;;
+    esac
+done
+
+LOGFILE="${LOGFILE:-$(mktemp -t gotestfmt.XXXXXX.log)}"
+echo >&2 "gotestfmt: Logging to $LOGFILE"
+STDERR_FILE=${LOGFILE%.log}.stderr
+STDOUT_FILE=${LOGFILE%.log}.stdout
+GOTESTFMT=${GOTESTFMT:-${GOPATH:-$HOME/go}/bin/gotestfmt}
+
+strip_go_downloading_lines() {
+    # Strip "go: downloading" lines which would cause jq (used by
+    # `strip_empty_coverage_lines`) to fail.
+    grep -v '^go: downloading'
+}
+
+strip_empty_coverage_lines() {
+    # This strips empty coverage lines from the output of `go test -cover -json`,
+    # which would otherwise be printed as error messages by gotestfmt.
+    jq -c 'select(.Output != null and (.Output | contains("coverage: 0.0%") | not))'
+}
+
+copy_to_logfile() {
+    # This copies the output of `go test -json` to the logfile. It also uses
+    # gotestfmt to format the output (but without GitHub Action workflow commands
+    # like `::group::` and `::endgroup::`) and removes ANSI color codes, to make
+    # the logfile more readable.
+    tee >(GITHUB_WORKFLOW='' "${GOTESTFMT}" -showteststatus | sed 's/\x1b\[[0-9;]*m//g' > "$LOGFILE")
```

Comment:
Because gotestfmt prints the `::group::` and `::endgroup::` GitHub Action workflow commands if that variable is set, and we don't want those in our logfile.

Reply:
mind rewording the comment a little bit to tell this is what triggers the GitHub Action workflow commands printing from `gotestfmt`?

Reply:
done